src/main/scala/org/labrad/concurrent/ExecutionContexts.scala

```diff
@@ -0,0 +1,27 @@
+package org.labrad.concurrent
+
+import java.util.concurrent._
+import java.util.concurrent.atomic.AtomicLong
+import scala.concurrent.{ExecutionContext, Future}
+
+object ExecutionContexts {
+
+  /**
+   * Create an ExecutionContext backed by a cached thread pool.
+   * All threads will be created in a group and named with the given name.
+   */
+  def newCachedThreadExecutionContext(name: String): ExecutionContext = {
+    val threadGroup = new ThreadGroup(name)
+    val threadFactory = new ThreadFactory {
+      val counter = new AtomicLong(0)
```

Comment:
Totally missed that -- yeah, I like that too. It's a very low expense for a useful feature :)